src/coord/src/session.rs

```diff
@@ -38,30 +38,32 @@ pub struct Session {
     prepared_statements: HashMap<String, PreparedStatement>,
     portals: HashMap<String, Portal>,
     transaction: TransactionStatus,
+    user: String,
     vars: Vars,
 }
 
 impl Session {
     /// Creates a new session for the specified connection ID.
-    pub fn new(conn_id: u32) -> Session {
+    pub fn new(conn_id: u32, user: String) -> Session {
         assert_ne!(conn_id, DUMMY_CONNECTION_ID);
-        Self::new_internal(conn_id)
+        Self::new_internal(conn_id, user)
     }
 
     /// Creates a new dummy session.
     ///
     /// Dummy sessions are intended for use when executing queries on behalf of
     /// the system itself, rather than on behalf of a user.
     pub fn dummy() -> Session {
-        Self::new_internal(DUMMY_CONNECTION_ID)
+        Self::new_internal(DUMMY_CONNECTION_ID, "system".into())
```

Comment:
will this cause an issue if someone wants to create a user named `system`?

Reply:
Good catch. Definitely not ideal. I fixed it up so that `pg_` and `mz_` role names are reserved for the system, just like they are for schemas, and then changed this to use `mz_system`. That should make it all fall out nicely.